federator/src/lib/Federator.js

```diff
@@ -23,41 +24,50 @@ module.exports = class Federator {
     }
 
     async run() {
-        try {
-            const currentBlock = await this.mainWeb3.eth.getBlockNumber();
-            const toBlock = currentBlock - this.config.confirmations || 120;
-            this.logger.info('Running to Block', toBlock);
-
-            if (toBlock <= 0) {
-                return false;
-            }
-
-            if (!fs.existsSync(this.config.storagePath)) {
-                fs.mkdirSync(this.config.storagePath);
-            }
-
-            let fromBlock = null;
+        let retries = 3;
+        const sleepAfterRetrie = 30000;
+        while(retries > 0) {
             try {
-                fromBlock = fs.readFileSync(this.lastBlockPath, 'utf8');
-            } catch(err) {
-                fromBlock = this.config.mainchain.fromBlock || 0;
-            }
-            fromBlock = parseInt(fromBlock)+1;
-            this.logger.debug('Running from Block', fromBlock);
+                const currentBlock = await this.mainWeb3.eth.getBlockNumber();
+                const toBlock = currentBlock - this.config.confirmations || 120;
+                this.logger.info('Running to Block', toBlock);
 
-            const logs = await this.mainBridgeContract.getPastEvents('Cross', {
-                fromBlock,
-                toBlock
-            });
-            if (!logs) return;
+                if (toBlock <= 0) {
+                    return false;
+                }
 
-            this.logger.info(`Found ${logs.length} logs`);
-            await this._processLogs(logs);
+                if (!fs.existsSync(this.config.storagePath)) {
+                    fs.mkdirSync(this.config.storagePath);
+                }
 
-            return true;
-        } catch (err) {
-            this.logger.error(new Error('Exception Running Federator'), err);
-            process.exit();
+                let fromBlock = null;
+                try {
+                    fromBlock = fs.readFileSync(this.lastBlockPath, 'utf8');
+                } catch(err) {
+                    fromBlock = this.config.mainchain.fromBlock || 0;
+                }
+                fromBlock = parseInt(fromBlock)+1;
+                this.logger.debug('Running from Block', fromBlock);
+
+                const logs = await this.mainBridgeContract.getPastEvents('Cross', {
+                    fromBlock,
+                    toBlock
+                });
+                if (!logs) return;
+
+                this.logger.info(`Found ${logs.length} logs`);
+                await this._processLogs(logs);
+
+                return true;
+            } catch (err) {
+                this.logger.error(new Error('Exception Running Federator'), err);
+                retries--;
```

Comment:
What happened with the library  I sent you :P ?

Reply:
i had to add it in every call, it was more suited to the UI where each request is independent, here its easier to  just start the process again.